### ISSUE TITLE
Sitemap.aspx is not longer accepted bij Google Search Console

### ISSUE THREAD

**Issue by wbonekamp** [dnnsoftware/Dnn.Platform#3277]
Open Google Search console and add new sitemap: domain.com/sitemap.aspx
Error: can't fetch (type unknown)

url sitemap.aspx is available in browser ans is a valid xml format
Tested the url in several sitemap validators and all pass

When you save the sitemap.aspx in sitemap.xml the sitemap is accepted and show the indexed pages

## Steps to reproduce
List the steps to reproduce the behavior:
1. Open Google Search console
2. Add sitemap
3. domain.com/sitemap.aspx
4. Save

## Error log
can't fetch (unknown type)

## Affected version
DNN 9.2.2
DNN 9.3.2

**Comment by thabaum** [comment]
can this be saved as an XML instead of aspx I believe that is the issue here as it has been a long time problem for versions ever since it was created in DNN as far back as I can remember.  This would be a good one to resolve.

**Comment by valadas** [comment]
Hmm, I never had an issue submitting sitemap.aspx not sure if google changed something recently I don't have any new site right now to submit but all my existing sites are using sitemap.aspx and get refreshed correctly.

**Comment by thabaum** [comment]
i can do it through the google webmaster site. but never using DNN

**Comment by wbonekamp** [comment]
Try a new DNN9 website and add your sitemap.aspx in Google Search Console. You wil see it's not accepted.

And yes sitemaps that were added in the past keep working on sitemap.aspx.
The problem only exist for new websites.

**Comment by valadas** [comment]
@thabaum 
> The format inside the file I believe is just like an XML file.

Yes not only the format but the mime-type too. I think what happens is that google now supports multiple sitemap formats and wrongly relies on the file name instead of the returned mime type to decide the type of sitemap.

**Comment by valadas** [comment]
For those who know more about it than me, could url routing (like used for service routes) be used in a similar fashion to point requests for SiteMap.xml to SiteMap.aspx transparently?

Something along the lines of:
```
RegisterRoutes(RouteCollection routes)
{
    routes.MapPageRoute(
        "sitemap",
        "sitemap.xml",
        "~/sitemap.aspx"
    );
}
```

**Comment by trouble2** [comment]
I know Sacha Trauwaen made the open url rewriter module which (among some other nice additions) took care of this: https://github.com/sachatrauwaen/OpenUrlRewriter

**Comment by mikesmeltzer** [comment]
@valadas I'm not sure if routing would work without trying it but this could be done using a URL Provider extension. I'm going to be doing something similar soon for a portal level robots.txt file.  I'll take a look at this if this is still open when I look at the robots and see if it would work in a similar way.

**Comment by wbonekamp** [comment]
Rewrite is a workaround option but I hoped it could be fixed in DNN core.

I assume many DNN users have this problem and the sitemap functionality is quite essential to get your DNN website indexed the right way

**Comment by dieterdtx** [comment]
I only just now found this thread, I was the one posting this on the Google Forum (since GSC does not have a real support). I will update my question there with the remark about file extension.

Thank you @EPTamminga for the rewrite solution/workaround.

**Comment by EdwardGraham** [comment]
Will this:

<rewrite>
	<rules>
		<rule name="SiteMap"
		 enabled="true"
		 patternSyntax="Wildcard"
		 stopProcessing="true">
			<match url="*sitemap.xml" />
			<action type="Rewrite"
			  url="{R:1}sitemap.aspx"
			  appendQueryString="true" />
		</rule>
	</rules>
</rewrite>

create an .xml file and if so where do I place it in web.config.  please forgive me but I am a newbie

**Comment by EPTamminga** [comment]
@EdwardGraham See my comment with my instructions placed here on 19 nov.

**Comment by EdwardGraham** [comment]
I saw that and my question was where to put it in the web.config

**Comment by EPTamminga** [comment]
In the <system.webServer> section.

Just google on “how to add redirect rule in web config” and you will find several instructions.

**Comment by EdwardGraham** [comment]
EPT Thank you it goes right under the </handles>   and seems to be working now off to google to try and register the site
Thank you

**Comment by mikesmeltzer** [comment]
While there is a workaround, given that Google Search Console is quite popular and a lot of solutions use .xml sitemaps, I think we should look at adding .xml support out of the box for this.

Please go ahead and assign it to me, I will take a look at a potential resolution for it in the coming weeks that doesn't require new core dependencies.

**Comment by EdwardGraham** [comment]
Thank you!


-----Original Message-----

**Comment by valadas** [comment]
Awesome @mikesmeltzer Assigned to you and triaged.

**Comment by wbonekamp** [comment]
Looks like Google Search Console does accept sitemap.aspx again

**Comment by mikesmeltzer** [comment]
> Looks like Google Search Console does accept sitemap.aspx again

Thanks for the info, glad this is no longer an impediment.

**Comment by andyduo** [comment]
Yes I was just successful in submitting the sitemap as well

**Comment by valadas** [comment]
Awesome, so was this a google bug and we close this issue or do we keep this because someone wanted to still implement a solution?

**Comment by Timo-Breumelhof** [comment]
IMO it would not be bad to server with with an XML extension, to prevent future issues

**Comment by EPTamminga** [comment]
We could add a standard rule in de web.config that is used in the install pack

**Comment by thabaum** [comment]
I think it should serve both... I thought it did in the past.  So if you type Sitemap.xml or Sitemap.aspx it would show the same file since... they are exactly the same.

Why not just create an alias for it?

**Comment by valadas** [comment]
Well, it is a bit more complex than an alias, xml is a static file that does not go through the normal pipeline of dynamic files. So we would need some kind of url handler to catch that url and pipe it through some dynamic handling code.

**Comment by mitchelsellers** [comment]
I personally would not support the addition of an additional alias to this file.  It should have one name, if we want to change to sitemap.xml we can, but that would be a breaking change for others.  There is no impact with regards to it being sitemap.aspx.

**Comment by Timo-Breumelhof** [comment]
> 
> 
> I personally would not support the addition of an additional alias to this file. It should have one name, if we want to change to sitemap.xml we can, but that would be a breaking change for others. There is no impact with regards to it being sitemap.aspx.

True, but Google expects the xml extension apparently (not completely unreasonable IMO). So I think it would be good to avoid issue in the future in case Google would change their "policy" again?

**Comment by EPTamminga** [comment]
A rule in web.config solves this case? No need to change sitemap.aspx?

**Comment by mitchelsellers** [comment]
Per the guidelines in google here https://support.google.com/webmasters/answer/183668?hl=en

There is not a requirement to have it named .xml, the format has to be one of many supported but the name isn’t noted as a requirement. 

Adding a redirect to web.config depends on an optional IIS component and could be a breaking change. It also adds a second redirect manager to all requests

**Comment by rodrigoratan** [comment]
If the .xml extension is really needed, why not put a button that generates the file? and if it already exists, warns the user and asks if he wants to replace

**Comment by jasclar** [comment]
why not just add "Sitemap: https://www.exampledomainhere.com/sitemap.aspx" into your robots.txt ?

**Comment by dieterdtx** [comment]
> why not just add "Sitemap: https://www.exampledomainhere.com/sitemap.aspx" into your robots.txt ?

There was a period (a couple of months) where Google Search Console did not accept the .aspx extension. In the end of March 2020 and onward GSC accepted it again.
